lib/ansible/modules/clustering/k8s/_kubernetes.py

```diff
@@ -349,7 +349,7 @@ def main():
         mutually_exclusive=(('file_reference', 'inline_data'),
                             ('url_username', 'insecure'),
                             ('url_password', 'insecure')),
-        required_one_of=(('file_reference', 'inline_data')),
+        required_one_of=(('file_reference', 'inline_data'),),
```

Comment:
I'd add a space after comma to be compliant with PEP8.
But for readability I'd also make it multiline:
```python
        required_one_of=(
            ('file_reference', 'inline_data'),
        ),
```

Reply:
Still I think this is the wrong fix and we should not be adding the comma here.
Ansible should simply handle this case correctly.

Reply:
Welcome in the real world, where rules of thumb are rules of thumb and not religious dogmas.

Here is another one: Robustness principle (Postel's Law): *Be liberal in what you accept, and strict in what you send.*
https://en.wikipedia.org/wiki/Robustness_principle

Reply:
Yes, there's Criticism section there.

I'm not in favor encouraging people to send random stuff into functions, which leads to misleading understanding of what it does to the code reader and created way more diversity over codebase.

Anyway @Akasurde has already suggested a good validation solution to this one.

Reply:
We're not sending random stuff into functions. We are making the interface (which is codified) accept more than just lists. Which is something we have done in multiple places already.

Reply:
Where did @Akasurde suggest an alternative ?

Reply:
API-wise it's a terrible idea to accept multiple argument types.

He suggested this in Slack and it's related to #39924. I think @Akasurde will update the PR once he has time.

Reply:
@webknjaz In life, it's a terrible idea to speak in absolutes. One day you'll figure that out...

In the meantime you can rewrite Ansible. I'd start with modifying everything in Ansible that accepts single types as a lists. That is *EVERY* YAML list as input. Good luck with that...